### ISSUE TITLE
Component in a column cell only only renders once ignoring filtering, sorting or pagination.

### ISSUE THREAD

**Issue by PKWebAppDev** [jbetancur/react-data-table-component#438]
Hello all.  

I've been playing around with the library and have hit a roadblock.  Now I've been looking through the documentation and past issues on the repo, and can't seem to find something hugely similar.  

Basically I have a column like so

`
const columns = [
. . .

    {
        name: 'Actions',
        selector: 'userid',
        sortable: true,
        cell: d => <div>

            {d.userid}
            <ColModalButton userid={d.userid} />

        </div>

    },
]
`
Basically for each row, I have an Actions column where I can press a button to open a Modal with a userid prop that is used to fetch pertinent data from an API.  

However, these ColModalButton components inside the cell, are only rendered _once_ during the initial render.  

So as a result, when you use Filtering, Sorting or Pagination, the order of ColModalButton's for Page 1 will not rerender to reflect changes and will persist no matter how the table is filtered/sorted/paged. 

For example, if my RDT's very first row is 225.

![image](https://user-images.githubusercontent.com/39636373/72856718-45023e00-3c70-11ea-82bb-7249eb67e696.png)

and I go to the next page, where that first row has userid 231,

![image](https://user-images.githubusercontent.com/39636373/72856792-7975fa00-3c70-11ea-82b6-4b5f6028807d.png)

If I press user 231's ModalButton, I still get 225's Modal render since it never rerendered after initialization.  

I know this is the culprit since I've disabled filtering,sorting and pagination and every modal displays the correct information through the API but then the table is too large to work with.   

What are the easiest options here?  Could I force a rerender maybe on "onSort" or "onChangePage"?

I define my RDT as such `const BasicTable = ({ data, col }) => {

    const [filterText, setFilterText] = React.useState('');
    const filteredItems = data.filter(item => item.fName.toLowerCase().includes(filterText.toLowerCase())
        || item.lName.toLowerCase().includes(filterText.toLowerCase())
        || item.email.toLowerCase().includes(filterText.toLowerCase())
    );

    const [filterCols, setFilterCols] = React.useState('');

    const filteredCols = columns

    const [highlight, setHighlight] = React.useState(false);
    return (
        <DataTable
            title="Water Club Members"
            columns={col}
            theme='dark'
            

            highlightOnHover={true}
            data={filteredItems}
            striped
        
            pagination

            conditionalRowStyles={conditionalRowStyles}
            
            subHeader
            subHeaderComponent={<Filter onFilter={value => setFilterText(value)} />}

        />
    );
};`

**Comment by jbetancur** [comment]
I would have to see all of your code or preferably a codesandbox.  I see you may be managing column state in React state which might be the culprit, but need to see all the code.

Here is a quick code sandbox where I just put some buttons with an onClick handler and sorting or paging and filtering shows the correct id.

https://codesandbox.io/s/react-data-table-sandbox-render-check-k8zki

**Comment by PKWebAppDev** [comment]
Hi John,

Been playing around with the sandbox trying to duplicate the issue..

But in the meantime I've actually found a (albeit non-elegant) "work-around" to my issue!  

Since columns (and thus the content of my Modal component) only renders once during the initial default table render,  

`try{ if (modalButtonisClicked) {`
`fetch('api/yourRoute/yourMethod?userid=' + userid) `
`.then(response => response.json())`
`                .then(resp => {`
`                    setInfos(resp);`
`                    setLoading(false);`
`                });`

I can just force the API to fire whenever my modalButtonisClicked hook is set to true
ensuring that the right data is displayed even after pagination/filtering.

**Comment by PKWebAppDev** [comment]
Hi John,


I'll try to produce my error in that sandbox you've supplied, but it is a bit hard since I had to use my own API.

In the meantime, basically what I ended up doing was essentially forcing a (albeit inelegant) work-around to my issue.

On the modal's open hook set to true, I fetch data on a modal by modal basis.  Of course, I'm fully aware of how this boggles performance as I scale up but for now, this does work for me.
